src/NGen.cpp

```diff
@@ -327,74 +331,99 @@ int main(int argc, char *argv[]) {
+    output_time_index.resize(keys.size());
+    std::fill(output_time_index.begin(), output_time_index.end(),0);
+
+    // now create the layer objects
+
+    // first make sure that the layer are listede in decreasing order
+    boost::range::sort(keys, std::greater<int>());
+
+    std::vector<std::shared_ptr<ngen::Layer> > layers;
+    layers.resize(keys.size());
+
+    for(long i = 0; i < keys.size(); ++i)
+    {
+      auto& desc = layer_meta_data.get_layer(keys[i]);
+      std::vector<std::string> cat_ids;
+
+      // make a new simulation time object with a differnt output interval
+      Simulation_Time sim_time(*manager->Simulation_Time_Object, time_steps[i]);
+  
+      for ( std::string id : features.catchments(keys[i]) ) { cat_ids.push_back(id); }
+      if (keys[i] != 0 )
+      {
+        layers[i] = make_shared<ngen::Layer>(desc, cat_ids, sim_time, features, catchment_collection, pdm_et_data, 0);
+      }
+      else
+      {
+        layers[i] = make_shared<ngen::SurfaceLayer>(desc, cat_ids, sim_time, features, catchment_collection, pdm_et_data, 0, nexus_subset_ids, nexus_outfiles);
+      }
+
+    }
+
     //Now loop some time, iterate catchments, do stuff for total number of output times
-    for(int output_time_index = 0; output_time_index < manager->Simulation_Time_Object->get_total_output_times(); output_time_index++) {
-      //std::cout<<"Output Time Index: "<<output_time_index<<std::endl;
-      if(output_time_index%100 == 0) std::cout<<"Running timestep "<<output_time_index<<std::endl;
-      std::string current_timestamp = manager->Simulation_Time_Object->get_timestamp(output_time_index);
-      for(const auto& id : features.catchments()) {
-        //std::cout<<"Running cat "<<id<<std::endl;
-        auto r = features.catchment_at(id);
-        //TODO redesign to avoid this cast
-        auto r_c = dynamic_pointer_cast<realization::Catchment_Formulation>(r);
-        r_c->set_et_params(pdm_et_data);
-        double response = r_c->get_response(output_time_index, 3600.0);
-        std::string output = std::to_string(output_time_index)+","+current_timestamp+","+
-                             r_c->get_output_line_for_timestep(output_time_index)+"\n";
-        r_c->write_output(output);
-        //TODO put this somewhere else.  For now, just trying to ensure we get m^3/s into nexus output
-        try{
-          response *= (catchment_collection->get_feature(id)->get_property("areasqkm").as_real_number() * 1000000);
-        }catch(std::invalid_argument &e)
+    for( int count = 0; count < manager->Simulation_Time_Object->get_total_output_times(); count++) 
+    {
+      // The Inner loop will advance all layers unless doing so will break one of two constraints
+      // 1) A layer may not preceed ahead of the master simulation objects current time
+      // 2) A layer may not preceed ahead of any layer that is computed before it
+      // The do while loop insures that all layers are tested at least once while allowing 
```

Comment:
```suggestion
      // The do while loop ensures that all layers are tested at least once while allowing 
```
@PhilMiller #bandwagon